airflow/hooks/dbapi.py

```diff
@@ -188,25 +206,21 @@ def run(self, sql, autocommit=False, parameters=None, handler=None):
                 self.set_autocommit(conn, autocommit)
 
             with closing(conn.cursor()) as cur:
-                results = []
                 for sql_statement in sql:
+                    results = []
                     self._run_command(cur, sql_statement, parameters)
                     if handler is not None:
                         result = handler(cur)
                         results.append(result)
```

Comment:
This doesn’t seem right.

Reply:
I'm confused a bit because this change is based on your comment 
https://github.com/apache/airflow/issues/23623#issuecomment-1122758517
I thought that all `hook.run()` methods should return the result from the last SQL statement

Reply:
Simply moving `results = []` into the loop is wrong because it makes `execute` always return a list with a single element, which is wrong. The correct behaviour is

1. If the input is a list of SQL queries, return a list containing results from each query.
2. If the input is a str containing one single SQL query, return the query’s result.
3. If the input is a str containing multiple queries, return the last query’s result.

The last one is the thing talked about in the linked comment, but this change also modifies the behaviour of the first scenario.

Reply:
I think that list is a good container and personally, for me, solution 1-ii seems the best one.
Let's say that we have the following list:
`sql=["INSERT ...;", "SELECT 1; SELECT 2;", "DELETE ...;", "SELECT 3:"]`

I think that returning should be:
`[None, [(2,)], None,[(3,)],]`

Also, the s 1-i is easy to implement if we will try to split every statement in the list.
But in this case, lengths of input and output lists might be different, which is not right thing for me

Reply:
If we agree with 3, I think 1-ii should be `[None, (2,), None, (3,)]` instead for consistency.

Reply:
Makes sense, and I think this matches the current behaviour implemented by various hooks (or at least most of them).